lib/db/queries.go

```diff
@@ -36,15 +38,61 @@ func (q *Queries) CreateOrUpdateRota(ctx context.Context, p CreateOrUpdateRotaPa
 		})
 	}
 	if err != nil {
-		var pgError *pgconn.PgError
-		if errors.As(err, &pgError) {
-			switch pgError.Code {
-			case "23505":
-				return "", ErrAlreadyExists
-			}
-		}
+		err = mapError(err)
 		l.Error("failed to save rota", zap.Error(err))
 		return "", err
 	}
+	err = q.updateMembersList(ctx, rotaId, p.Members)
+	if err != nil {
+		err = mapError(err)
+		l.Error("failed to save rota members", zap.Error(err))
+		return "", err
+	}
 	return rotaId, nil
 }
+
+func (q *Queries) updateMembersList(ctx context.Context, rotaId string, members []Member) error {
+	l := zapctx.Logger(ctx)
+	e, err := q.ListUserIDsByRotaID(ctx, rotaId)
+	if err != nil {
+		l.Error("unable_to_fetch_existing_members", zap.Error(err))
+		return err
+	}
+	for _, userId := range e {
+		inx := slices.IndexFunc(members, func(member Member) bool { return member.UserID == userId })
+		if inx == -1 {
+			err = q.deleteMember(ctx, userId)
+			if err != nil {
+				l.Error("unable_to_fetch_existing_members", zap.Error(err))
+				return err
+			}
+		}
+	}
+	for _, m := range members {
+		_, err = q.saveMember(ctx, saveMemberParams{
+			RotaID:   rotaId,
+			UserID:   m.UserID,
+			Metadata: m.Metadata,
+		})
+		if err != nil {
+			err = mapError(err)
+			if errors.Is(err, ErrAlreadyExists) {
+				continue
+			} else {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func mapError(err error) error {
+	var pgError *pgconn.PgError
+	if errors.As(err, &pgError) {
+		switch pgError.Code {
+		case "23505":
```

Comment:
The magic list

https://www.postgresql.org/docs/current/errcodes-appendix.html